Managers/MovableMan.cpp

```diff
@@ -1046,37 +1087,11 @@ bool MovableMan::ValidateMOIDs() {
 // Method:          ValidMO
 //////////////////////////////////////////////////////////////////////////////////////////
 // Description:     Indicates whether the passed in MovableObject pointer points to an
-//                  MO that's currently active in the simulation, and kept by this
-//                  MovableMan. Internal optimization is made so that the same MO can
-//                  efficiently be checked many times during the same frame.
+//                  MO that's currently active in the simulation, and kept by this MovableMan.
 
 bool MovableMan::ValidMO(const MovableObject *pMOToCheck)
 {
-    bool found = false;
-
-    if (pMOToCheck)
-    {
-
-        // See if this MO has been found earlier this frame
-        for (auto itr = m_ValiditySearchResults.begin(); !found && itr != m_ValiditySearchResults.end(); ++itr)
-        {
-            // If the MO is found to have been searched for earlier this frame, then just return the search results
-            if (itr->first == pMOToCheck)
-                return itr->second;
-        }
-
-        // Check actors
-        found = found ? true : IsActor(pMOToCheck);
-        // Check Items
-        found = found ? true : IsDevice(pMOToCheck);
-        // Check particles
-        found = found ? true : IsParticle(pMOToCheck);
-
-        // Save search result for future requests this frame
-        m_ValiditySearchResults.push_back(std::pair<const MovableObject *, bool>(pMOToCheck, found));
-    }
-
-    return found;
+    return pMOToCheck && pMOToCheck->GetID() != g_NoMOID;
```

Comment:
Pretty sure this works fine. I’ve not seen any issues whatsoever while playing, and I’ve tested these changes a lot. That aside, from a purely philosophical perspective an MO can only be a valid MO if it has a MOID, so whatever behavioural differences this may introduce seems inarguably more accurate.